packages/web3-eth-contract/test/integration/contract_erc721.test.ts

```diff
@@ -70,6 +71,15 @@ describe('contract', () => {
 				contractDeployed = await contract.deploy(deployOptions).send(sendOptions);
 			});
 
+			const getTokenId = async (res: Receipt) => {
+				const topic = (res.logs as EventLog[])[0]?.topics[0];
+				const logs = await contractDeployed.getPastEvents('Transfer');
+
+				return toBigInt(
+					(logs.find(l => (l as EventLog).topics.includes(topic)) as EventLog)
```

Comment:
It seems that this error: https://github.com/web3/web3.js/actions/runs/3269305208/jobs/5386153617 happens when the value is undefined. 

So, do you think it is a good solution to try to fetch the past event of 'Transfer' again if  `tokenId` was undefined?